torch_geometric/data/data.py

```diff
@@ -692,6 +712,44 @@ def num_faces(self) -> Optional[int]:
             return self.face.size(self.__cat_dim__('face', self.face))
         return None
 
+    # :obj:`FeatureStore` interface ###########################################
+
+    def _put_tensor(self, tensor: FeatureTensorType, attr: TensorAttr) -> bool:
+        r"""Stores a feature tensor in node storage."""
+
+        if not attr.is_set('index'):
+            attr.index = None
+
+        out = getattr(self, attr.attr_name, None)
+        if out is not None:
+            # Attr name exists, handle index:
+            out[attr.index] = tensor
+        else:
+            # No attr nane, just store tensor:
+            setattr(self, attr.attr_name, tensor)
+        return True
+
+    def _get_tensor(self, attr: TensorAttr) -> Optional[FeatureTensorType]:
+        r"""Obtains a feature tensor from node storage."""
+        # Retrieve tensor and index accordingly:
+        tensor = getattr(self, attr.attr_name)
+        if tensor is not None:
+            # TODO this behavior is a bit odd, since TensorAttr requires that
+            # we set `index`. So, we assume here that indexing by `None` is
+            # equivalent to not indexing at all, which is not in line with
+            # Python semantics.
+            return tensor[attr.index] if attr.index is not None else tensor
```

Comment:
One important thing: I think we need to make use of `cat_dim` here to know which in which index to select. See [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/loader/utils.py#L103-L104) for an example:
```python
cat_dim = self.__cat_dim__(attr.attr_name, tensor)
```

Reply:
I imagined that it would be on the user to provide the correct slice for this; for example,
```
(*([slice(None)] * (cat_dim - 1 if cat_dim > 0 else cat_dim)), attr.index)
```
but I'm happy to implement it this way as well.

Reply:
I see. That makes sense.